pkgs/development/libraries/qt-5/5.15/default.nix

```
@@ -59,14 +59,22 @@ let
         rev = "a059e7404a6db799f4da0ad696e65ae9c854b4b0";
       in
       {
-        version = "${branchName}-${lib.substring 0 8 rev}";
+        version = "${branchName}-${lib.substring 0 7 rev}";
 
         src = fetchgit {
           url = "https://github.com/qt/qtwebengine.git";
-          sha256 = "19l1i4sk3pvnwbvz5s97jchqawfz8k1xmjza29bgvkp1zz96r0jx";
+          sha256 = "0646fk8gg6lcsp1c50is9jcvr7yifn508709jgds90cnvzzb42mx";
           inherit rev branchName;
           fetchSubmodules = true;
           leaveDotGit = true;
+          name = "qtwebengine-${substring 0 7 rev}.tar.gz";
+          postFetch = ''
+            # remove submodule .git directory
+            rm -rf $out/src/3rdparty/.git
+            # compress to not exceed the 2GB output limit
+            mv $out source
+            tar -I 'gzip -n' -cf $out source
```

```suggestion
            tar --sort=name -I 'gzip -n' -cf $out source
```
should hopefully help a bit.

---

Yep, there are several source of indeterminism: files order, timestamps, uid/gids, file modes and compression. Thankfully the debian maintainers already figured out most of these: 

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=647522
https://wiki.debian.org/ReproducibleBuilds/TimestampsInTarball
https://lists.nongnu.org/archive/html/help-tar/2015-05/msg00006.html

---

I am not sure which are important and which parts nix already did.